backend/api/request_create.go

```
@@ -0,0 +1,102 @@
+
+	if _, err := server.queries.GetCommunity(ctx, req.CommunityID); err != nil {
+		if err == sql.ErrNoRows {
+			err := errors.New("No community exists with given ID.")
+			ctx.JSON(http.StatusNotFound, errorResponse(err))
+		} else {
+			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+		}
+		return
+	}
+
+	if len(req.Items) == 0 {
+		err := errors.New("Request must contain at least one item.")
+		ctx.JSON(http.StatusNotFound, errorResponse(err))
+		return
+	}
+
+	arg := db.CreateRequestParams{
+		UserID:      authPayload.UserID,
+		CommunityID: sql.NullInt64{Int64: req.CommunityID, Valid: true},
+	}
+	if req.StoreID != nil {
+		if _, err := server.queries.GetStore(ctx, *req.StoreID); err != nil {
+			if err == sql.ErrNoRows {
+				err := errors.New("No store exists with given ID.")
+				ctx.JSON(http.StatusNotFound, errorResponse(err))
+			} else {
+				ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+			}
+			return
+		}
+
+		arg = db.CreateRequestParams{
+			UserID:      authPayload.UserID,
+			CommunityID: sql.NullInt64{Int64: req.CommunityID, Valid: true},
+			StoreID:     sql.NullInt64{Int64: *req.StoreID, Valid: true},
+		}
+	}
+
+	request, err := server.queries.CreateRequest(ctx, arg)
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+		return
+	}
+
+	for _, item := range req.Items {
+		arg := db.CreateItemParams{
+			Name:           item.Name,
+			RequestedBy:    authPayload.UserID,
+			RequestID:      request.ID,
+			QuantityType:   item.QuantityType,
+			Quantity:       item.Quantity,
+			PreferredBrand: sql.NullString{String: item.PreferredBrand, Valid: len(item.PreferredBrand) > 0},
+			Image:          sql.NullString{String: item.Image, Valid: len(item.Image) > 0},
+			ExtraNotes:     sql.NullString{String: item.ExtraNotes, Valid: len(item.ExtraNotes) > 0},
+		}
+
+		_, err := server.queries.CreateItem(ctx, arg)
+		if err != nil {
```

Let's ignore errors here, we don't want to return from the handler while items are being added.

---

Ah yes good catch